### DISCUSSION TITLE
[8.0 QA] New Layout

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25688] [Announcement]
#### Feature owner: @cdedreuille 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by JReinhold** [comment]
**Project**: None - but visible in [this sandbox](https://stackblitz.com/edit/github-t8esdj?file=.storybook/manager.jsx&preset=node)
**Version**: `8.0.0-alpha.12`
**Bug/feedback**: Custom (addon) tabs are missing from the toolbar.

Any custom tabs added via the register-addon API are not shown in the toolbar as they used to be.

See this screenshot from 7.6:
![image](https://github.com/storybookjs/storybook/assets/5678122/88d38e94-5e83-4f75-8e37-9447384c941f)

Compare with 8.0:
![image](https://github.com/storybookjs/storybook/assets/5678122/c725f49a-1fae-418e-a1d5-bfe24b37d1e7)

It's also possible that this is not a layout issue but the API to register addon tabs have a change or bug.

Here's an 8.0 sandbox demonstrating the issue:
https://stackblitz.com/edit/github-t8esdj?file=.storybook%2Fmanager.jsx

And the exact same code in a 7.6 sandbox showing the tab:
https://stackblitz.com/edit/github-sibamb?file=.storybook%2Fmanager.jsx

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25758.

**Comment by valentinpalkovic** [comment]
**Project**: Bitwarden (Angular)
**Version**: 8.0.0-alpha.13
**Bug/feedback**: Resizing of Storybook Panel doesn't always apply
**Tested Browser**: Edge (but also encountered it in Chrome a while ago)

When the panel is resized fast, the changes aren't always applied:

https://github.com/storybookjs/storybook/assets/5889929/73c3edf4-40bd-41a7-abda-887ffbb2c60f

Reproduction:

1. Go to https://stackblitz.com/edit/github-t8esdj?file=.storybook%2Fmanager.jsx
2. Open the `Primary` Button Story.
3. Try to resize the Storybook Panel fast

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25759

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

When activating the blurred vision filter from the accessibility addon, the preview area shows weird borders:

Storybook 7
<img src="https://github.com/storybookjs/storybook/assets/1671563/97a44c57-a89a-4a12-9703-f4df82474100" width="400" />

Storybook 8
<img src="https://github.com/storybookjs/storybook/assets/1671563/e6449c8b-d1d4-431d-83da-f7df445f937b" width="400" />

This is caused because of this newly added box-shadow in the preview iframe:
https://github.com/storybookjs/storybook/blob/next/code/ui/manager/src/components/preview/Iframe.tsx#L16 


**Repro:**
1. Go to https://stackblitz.com/edit/github-csg7tu
2. Open any story
3. Use the blurred vision filter from addon-a11y

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25818

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

It seems that not all parts of Storybook are using the new icons, so the browser console gets many deprecation warnings when older icons are being used such as:

```js
Use of deprecated props in the button component detected, see the migration notes at https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#new-ui-and-props-for-button-and-iconbutton-components

Use of the deprecated Icons (check) component detected. Please use the @storybook/icons component directly. For more informations, see the migration notes at https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#icons-is-deprecated

Use of the deprecated Icons (play) component detected. Please use the @storybook/icons component directly. For more informations, see the migration notes at https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#icons-is-deprecated
```

**Scenario 1:**

When using addon-interactions and accessing a story that contains a play function. The addon interactions panel seems to be using old icons (check, play).

**Scenario 2:**

When creating your own toolbar ([from official docs](https://storybook.js.org/docs/essentials/toolbars-and-globals#global-types-and-the-toolbar-annotation)), you can define icons via their name like so:

```ts
const preview = {
  //... other properties,
  globalTypes: {
    theme: {
      name: 'Theme',
      description: 'Theme for the components',
      defaultValue: 'light',
      toolbar: {
        icon: 'circlehollow',
        items: [
          { value: 'light', icon: 'circlehollow', title: 'light' },
          { value: 'dark', icon: 'circle', title: 'dark' },
          { value: 'side-by-side', icon: 'sidebar', title: 'side by side' },
        ],
      },
    },
  },
};

export default preview;
```

When clicking on the newly created toolbar button, you will get as many warnings as the amount of icons you create:
<img width="600" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/d4925d79-cf0a-44a9-ae87-6d7381bf9c7f">


**Repro:**
1. Go to https://stackblitz.com/edit/github-csg7tu
2.1 Open the `Page - Logged In` story
2.2 Click on the circle toolbar
3. See the warnings

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25760

**Comment by vanessayuenn** [comment]
**Project**: Chromatic
**Version**: v8.0.0-alpha.14
**Bug/feedback:** 

1. Keyboard shortcut still show up as the old ones, but when you click it, it changes to the correct combination with ⎇.
2.  did we change search to use `cmd + k`? it still shows up as `/` here, and search is also triggered by `/` 

<img width="596" alt="Screenshot 2024-01-24 at 22 38 34" src="https://github.com/storybookjs/storybook/assets/6842965/f0e89a0e-8848-4fdf-a2b5-d3a86bdca121">

**Reply by JReinhold** [reply]
I experienced the same thing, but I think this is cached in `localStorage`, which is why it stays the old until you clear it maybe? @cdedreuille

**Reply by ndelangen** [reply]
Yes, it's cached, if you press the reset to defaults button at the bottom you get the new shortcuts.

**Reply by cdedreuille** [reply]
Yes this is specified in the migration notes. As they are cached you to either clear your cache or press the reset button.

**Comment by vanessayuenn** [comment]
**Project**: Chromatic
**Version**: v8.0.0-alpha.14
**Bug/feedback:** 

It's possible to resize the side bar to a point where the gear icon gets cut off

<img width="296" alt="image" src="https://github.com/storybookjs/storybook/assets/6842965/1ce35d85-26c8-40e7-bce3-bfb45be15c84">

**Reply by JReinhold** [reply]
@cdedreuille I'd expect this to be the composition header issue we discussed.

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25817

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

When selecting a story that is placed at the bottom of the sidebar, then toggling the addon panel, the panel jumps to the bottom:

![2024-01-25 15 06 28](https://github.com/storybookjs/storybook/assets/1671563/3f75d914-28e6-4687-ad4f-8b16eaf03915)

**Repro:**
1. Go to https://5f84c5baa35bdd0022a7684f-wvwojnmlga.chromatic.com/?path=/story/components-topbanner--default
2. Toggle addon panel/change its position

**Comment by ndelangen** [comment]
On the mobile view we should not hint at keyboard shortcuts, the search bar has the keyboard shortcut indicator
![Screenshot 2024-01-25 at 14 38 03](https://github.com/storybookjs/storybook/assets/3070389/270731fc-8504-4414-85ae-52314de42e46)

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25816

**Comment by valentinpalkovic** [comment]
Project: Bitwarden
Version: 8.0.0-alpha.14
Bug/feedback: Ctrl + K triggers the browser search instead of Storybook's search

When a Story is focused and `CTRL + K` combination is pressed, the browsers search bar is focused. As discussed with @cdedreuille, we might revert the keyboard mapping and stick with the old `/` key instead

**Reply by vanessayuenn** [reply]
tracked https://github.com/storybookjs/storybook/issues/25815

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.14
**Bug/feedback:**

I don't know how bad this one is, as I was only able to reproduce it when first loading a Chromatic published Storybook (after it won't happen), but it's quite weird. When you open a Storybook for the first time, if it lands on a docs page (I assume), the preview container is shorter than it should be. I think it's potentially because it's saving room for the addons panel, even though the addons panel will never exist for a docs page.

Check this gif, and notice that about 15% of the width (right side) is white:

![2024-01-26 10 10 02](https://github.com/storybookjs/storybook/assets/1671563/7b4dd2fb-9b95-47b6-9f23-aecd6c4a8cb9)


The reason I assume it has something to do with addon panel logic is because when you visit a normal story, the panel takes the same width on the right side:
<img width="620" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/728b460b-c4ea-49eb-96ef-657bfdafa2f8">


**Repro:**
It's hard because it will only happen once.
1. Visit any deployed Storybook 8, here's an example: https://5f84c5baa35bdd0022a7684f-koohzbxsar.chromatic.com/?path=/docs/getting-started-with-mealdrop--docs

**Comment by valentinpalkovic** [comment]
Project: React Vite Yarn PnP
Version: 8.0.0-alpha.15
Bug/feedback: @storybook/addon-viewport tried to access @storybook/icons, but it isn't declared in its dependencies.

`@storybook/addon-viewport` does not have `@storybook/icons` defined as a `dependency`, but rather as a `devDependency`. It seems, that the package isn't prebundled. Question. Should we make sure, that `@storybook/icons` get prebundled, or should it be a regular dependency?

**Reply by valentinpalkovic** [reply]
Fix: https://github.com/storybookjs/storybook/pull/25768

**Reply by cdedreuille** [reply]
@ndelangen I believe you are the best person to answer this question

**Reply by valentinpalkovic** [reply]
I can't replicate it anymore

**Reply by ndelangen** [reply]
I'd need to know how this happened:

> @storybook/addon-viewport tried to access @storybook/icons

Because what should happen is the builder-manager globalizes `@storybook/icons` when building.
I'd need more detailed steps to reproduce to understand what happened wrong.

I looked at this with @valentinpalkovic yesterday and couldn't demonstrate the problem.

**Comment by yannbf** [comment]
Project: React Vite
Version: 8.0.0-rc.2
Repro: https://stackblitz.com/edit/github-gxrjm4?file=.storybook%2Fmain.js&preset=node

When using an addon that provides a "tabs mode", such as @storybook/addon-designs, the tabs bar seems to have a slight misalignment (look at Canvas, Design):

<img width="705" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/1f6e0e38-832c-4e56-899e-977497ce6fa5">

Very minor, but I bet @cdedreuille would like to know about it

**Reply by cdedreuille** [reply]
Yes I know about this one but I think that's ok for now. I personally don't think the blue line should touch the left side.

**Comment by yannbf** [comment]
Very minor, but the "run tests" button floats atop the scrollbar (Storybook 8.1.0):
 
![image](https://github.com/storybookjs/storybook/assets/1671563/85725e6a-05d8-4a61-8bc7-f4313c2dd7db)

**Reply by cdedreuille** [reply]
Where is the settings icon?

**Reply by yannbf** [reply]
Good question, maybe it was a big bug altogether. But still, the circle thingy shouldn't have a z-index higher than the scrollbar

**Reply by domyen** [reply]
I can reproduce this too. In my large Storybook (2000 stories + composition), the ⚙️ settings and ▶️ play buttons get hidden when I narrow the sidebar just a little. 

I believe this is because long list items (aka the stories) don't wrap when the sidebar is narrow for some reason.
